packages/cardano-graphql-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/index.ts

```diff
@@ -0,0 +1,167 @@
+import { Cardano, StakePoolQueryOptions, StakePoolSearchProvider } from '@cardano-sdk/core';
+import { Logger, dummyLogger } from 'ts-log';
+import { Pool, QueryResult } from 'pg';
+import { PoolHashModel } from './types';
+import { mapPoolHash, toCoreStakePool } from './mappers';
+
+export class DbSyncStakePoolSearchProvider implements StakePoolSearchProvider {
+  #db: Pool;
+  #logger: Logger;
+
+  constructor(db: Pool, logger = dummyLogger) {
+    this.#logger = logger;
+    this.#db = db;
+  }
+
+  private async queryRetirements(hashesIds: number[]) {
+    const result = await this.#db.query(
+      `
+    SELECT  
+      tx.hash
+    FROM pool_retire pr
+    JOIN tx
+      ON tx.id = pr.announced_tx_id;
+    WHERE pr.hash_id IN $1
+    `,
+      [hashesIds]
+    );
+    return result.rows;
+  }
+  private async queryRegistrations(hashesIds: number[]) {
+    const result = await this.#db.query(
+      `
+    SELECT  
+      tx.hash
+    FROM pool_update pu
+    JOIN tx
+      ON tx.id = pu.registered_tx_id;
+    WHERE pu.hash_id IN $1
+    `,
+      [hashesIds]
+    );
+    return result.rows;
+  }
+  private async queryPoolRelays(updatesIds: number[]) {
+    // missing Relay hostname
+    const result = await this.#db.query(`
+    SELECT
+      relay.update_id as "updateId",
+      relay.ipv4 as ipv4,
+      relay.ipv6 as ipv6,
+      relay.port as port,
+      relay.dns_name as "dnsName"
+    FROM pool_update pr
+    WHERE update_id IN (${updatesIds})
+    `);
+    return result.rows;
+  }
+  private async queryPoolOwners(hashesIds: number[]) {
+    const result = await this.#db.query(
+      `
+    SELECT 
+      stake.view AS address 
+    FROM pool_owner owner
+    JOIN stake_address stake
+      ON stake.id = owner.addr_id
+    WHERE owner.pool_hash_id IN $1
+    `,
+      [hashesIds]
+    );
+    return result.rows;
+  }
+  private async queryPoolRewards(hashesIds: number[], limit?: number) {
+    const params: [number[] | number] = [hashesIds];
+    if (limit) params.push(limit);
+    const result = await this.#db.query(
+      `
+      WITH pool_epoch_reward as (
+        SELECT
+          COUNT(r.amount) as total_rewards,
+          pool_id,
+          r.spendable_epoch as epoch
+        FROM reward r
+        WHERE pool_id IN $1
+        GROUP BY pool_id, spendable_epoch
+        ORDER BY spendable_epoch DESC
+        ${limit ? 'LIMIT = $2' : ''}
+      )
+      SELECT 
+        per.*,
+        epoch.blk_count 
+      FROM pool_epoch_reward per
+      JOIN epoch 
+        ON epoch."no" = per.epoch;
+    `,
+      params
+    );
+    return result.rows;
+  }
+  private async queryLastPoolUpdate(hashId: number) {
+    // Missing PoolParameters.metadata
```

Comment:
@rhyslbw will we need to use smash directly? You mentioned something about dbsync using smash.

Reply:
https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/schema.md#pool_offline_data

SMASH has been [ported](https://github.com/input-output-hk/cardano-db-sync/blob/master/cardano-smash-server/CHANGELOG.md#1200) to the `cardano-db-sync` codebase, so for our purpose right now just consider we're running our own instance and the data should be there.